Server/mods/deathmatch/logic/CResource.cpp

```diff
@@ -1545,37 +1576,53 @@ bool CResource::ReadIncludedFiles(CXMLNode* pRoot)
 
             if (!strFilename.empty())
             {
-                std::string strFullFilename;
                 ReplaceSlashes(strFilename);
 
-                if (IsFilenameUsed(strFilename, true))
+                if (!IsValidFilePath(strFilename.c_str()))
                 {
-                    CLogger::LogPrintf("WARNING: Ignoring duplicate client file in resource '%s': '%s'\n", m_strResourceName.c_str(), strFilename.c_str());
-                    continue;
+                    m_strFailureReason = SString("Couldn't find file(s) %s for resource %s\n", strFilename.c_str(), m_strResourceName.c_str());
+                    CLogger::ErrorPrintf(m_strFailureReason);
+                    return false;
                 }
 
-                bool           bDownload = true;
-                CXMLAttribute* pDownload = Attributes.Find("download");
+                std::vector<std::string> vecFiles = GetFilePaths(strFilename.c_str());
 
-                if (pDownload)
+                if (vecFiles.empty())
                 {
-                    const char* szDownload = pDownload->GetValue().c_str();
-
-                    if (!stricmp(szDownload, "no") || !stricmp(szDownload, "false"))
-                        bDownload = false;
+                    m_strFailureReason = SString("Couldn't find file(s) %s for resource %s\n", strFilename.c_str(), m_strResourceName.c_str());
+                    CLogger::ErrorPrintf(m_strFailureReason);
+                    return false;
                 }
 
-                // Create a new resourcefile item
-                if (IsValidFilePath(strFilename.c_str()) && GetFilePath(strFilename.c_str(), strFullFilename))
+                for (const std::string& strFilePath : vecFiles)
                 {
-                    m_ResourceFiles.push_back(new CResourceClientFileItem(this, strFilename.c_str(), strFullFilename.c_str(), &Attributes, bDownload));
-                }
-                else
-                {
-                    m_strFailureReason = SString("Couldn't find file %s for resource %s\n", strFilename.c_str(), m_strResourceName.c_str());
-                    CLogger::ErrorPrintf(m_strFailureReason);
-                    return false;
+                    std::string strFullFilename;
+
+                    if (IsFilenameUsed(strFilePath, true))
+                    {
+                        CLogger::LogPrintf("WARNING: Ignoring duplicate client file in resource '%s': '%s'\n", m_strResourceName.c_str(), strFilePath.c_str());
+                        continue;
+                    }
+
+                    bool           bDownload = true;
+                    CXMLAttribute* pDownload = Attributes.Find("download");
+
+                    if (pDownload)
+                    {
+                        const char* szDownload = pDownload->GetValue().c_str();
```

Comment:
`pDownload->GetValue()` returns `std::string`. It's worth to use it instead doing of C-style string comparison via `stricmp`.

Reply:
Yes, I know it was before. But I think using modern C++ never hurts.